include-fragment-element.js

```diff
@@ -1,105 +1,105 @@
 /* eslint-disable github/no-then */
 
-;(function() {
-  'use strict'
-
-  const privateData = new WeakMap()
+export class IncludeFragmentElement extends HTMLElement {
+  constructor() {
+    super()
+    this._privateData = new WeakMap()
+    // Preload data cache
+    this._getData()['catch'](function() {
+      // Ignore `src missing` error on pre-load.
+    })
+  }
 
-  function fire(name, target) {
+  _fire(name, target) {
```

Comment:
I kind of prefer the free floating private module functions because there's no way to call them on elements in the tree. The underscore naming convention is an agreement to not call them, but who knows what browser extension code will come to rely on. I just assume that if a method is visible on an element it will be called by someone.